examples/cosmos/ics20/ics20.qnt

```diff
@@ -0,0 +1,226 @@
+// -*- mode: Bluespec; -*-
+
+/**
+ * A specification ICS20 fungible token transfer protocol.
+ *
+ * For now, this only models the denomination trace changes between transfers.
+ *
+ * Gabriela Moreira, Informal Systems, 2023
+ */
+module denomTrace {
+  type Hop = {
+    port: str,
+    channel: str,
+  }
+
+  type DenomTrace = {
+    path: List[Hop],
+    baseDenom: str,
+  }
+
+  pure def senderIsSource(sourcePort: str, sourceChannel: str, denomTrace: DenomTrace): bool =
+    denomTrace.path.length() >= 1 and
+    denomTrace.path.head() == { channel: sourceChannel, port: sourcePort }
+
+  pure def receiverIsSource(sourcePort: str, sourceChannel: str, denomTrace: DenomTrace): bool =
+    not(senderIsSource(sourcePort, sourceChannel, denomTrace))
```

Comment:
Related to [Igor's comment](https://github.com/informalsystems/quint/pull/778/files#r1159521018), I think these got mixed up – if the denom trace is prefixed with the `{sourcePort}/{sourceChannel}`, it means the packet arrived through this channel and the sending chain is a sink.

Reply:
Right! I was just trying to understand what I got wrong here, and yep, I switched those up. Fix incoming!

Reply:
WDYT of the name `goingTowardsOrigin` [here](https://github.com/informalsystems/quint/pull/778/commits/9fa9b8fa3102af9f682878b9d106bfd868fbf726)? I find this much easier to reason about (thanks @konnov for the name suggestion).

Reply:
Great idea! Maybe even `returningToOrigin`?

Reply:
I like `returningToOrigin`, but also feel like `goingTowardsOrigin` is more "to the point".  Returning can be seen as something about the whole trace, while this is about a single step: while it might be going towards the origin now, it might go in a different path in the next step and never return to the origin at all.

Reply:
Hmm, that makes sense. This is actually not completely local, as it is looking at where it came from. `returning` makes a lot of sense now :smile:

Reply:
What about `isBacktracking`?

Reply:
`isBacktracking` is good, I like it. Another idea: the [comment](https://github.com/cosmos/ibc-go/blob/457095517b7832c42ecf13571fee1e550fec02d0/modules/apps/transfer/keeper/relay.go#L18-L49) in the reference implementation describes this in terms of going forward or going back in the token timeline. I think this is a good way of explaining it, and if we can find a name that relates to that, it would be more coherent with the existing material. I guess `isBacktracking` is already close to that. Maybe something like `movingBackInTimeline`?

Reply:
I didn't grok the timeline explanation the first time, because physical time is still going forward while a token is passing backward. So the mention of time confused me.
In general, I prefer this approach over the source/sink nomenclature though.

We're on the right track here 😄

A very verbose option would be `movingBackAlongDenomTrace`?

Reply:
Good point and good suggestions. Maybe we can remove `Denom` and use `movingBackAlongTrace` since, in the spec context, trace can only be a denom trace.